pyccel/parser/semantic.py

```diff
@@ -830,14 +831,31 @@ def _handle_function(self, expr, func, args, **settings):
                 errors.report("Too many arguments passed in function call",
                         symbol = expr,
                         severity='fatal')
+
+            func_args = func.arguments if isinstance(func, FunctionDef) else func.functions[0].arguments
+            # Sort arguments to match the order in the function definition
+            input_no_kwargs = [a for a in args if a.keyword is None]
+            nargs = len(input_no_kwargs)
+            input_kwargs = []
+            for ka in func_args[nargs:]:
+                key = ka.name
+                relevant_args = [a for a in args[nargs:] if a.keyword == key]
+                assert len(relevant_args) <= 1
+                if len(relevant_args) == 0 and ka.has_default:
+                    input_kwargs.append(ka.default_call_arg)
+                elif len(relevant_args) == 1:
+                    input_kwargs.append(relevant_args[0])
+
+            args = input_no_kwargs + input_kwargs
```

Comment:
I wonder if this sorting could become potentially very expensive in a code containing many function calls with many arguments.

I don't know if this makes much of a difference, but we could use a single `input_args` list instead of two separate lists, in order to avoid the final concatenation. The code will still be readable and clear if we add some comments explaining what we are doing.

In any case we should make sure that `len(relevant_args)` is only called once. Now it is called 2 or 3 times.

Reply:
Fixed [416f118](https://github.com/pyccel/pyccel/pull/1210/commits/416f118a897982d506df1a90a15ea26730f076e4) and [096dcad](https://github.com/pyccel/pyccel/pull/1210/commits/096dcad82286a9ef0089e921c585f6291a6c33ba)